catalyst_voices/lib/widgets/containers/roles_chooser_panel.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:catalyst_voices/widgets/cards/role_chooser_card.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:flutter/widgets.dart';
+
+/// A panel that displays a series of [RoleChooserCard] widgets for selecting
+/// various account roles (e.g., voter, proposer, drep).
+class RolesChooserPanel extends StatelessWidget {
+  /// A map where keys are [AccountRole] enums and values are booleans
+  /// representing whether the corresponding role is selected.
+  final Map<AccountRole, bool> value;
+
+  /// A map similar to [value], indicating which roles
+  /// should be locked in their default state.
+  final Map<AccountRole, bool>? lockedValuesAsDefault;
+
+  /// A callback triggered when any role selection changes, passing
+  /// the updated map of role values.
+  final ValueChanged<Map<AccountRole, bool>>? onChanged;
+
+  /// A callback triggered when the user taps "Learn More" for a
+  /// specific role.
+  final void Function(AccountRole)? onLearnMore;
+
+  const RolesChooserPanel({
+    super.key,
+    required this.value,
+    this.lockedValuesAsDefault,
+    this.onChanged,
+    this.onLearnMore,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        RoleChooserCard(
```

Comment:
try doing smth like this so you don't need to update this widget when `AccountRole` changes.

```dart
    return Column(
      mainAxisSize: MainAxisSize.min,
      children: AccountRole.values
          .map<Widget>(
            (role) {
              return RoleChooserCard(
                imageUrl: VoicesAssets.images.roleVoter.path,
                value: value[AccountRole.voter] ?? false,
                label: context.l10n.voter,
                lockValueAsDefault:
                    lockedValuesAsDefault?[AccountRole.voter] ?? false,
                onChanged: (newValue) {
                  onChanged?.call({...value, AccountRole.voter: newValue});
                },
                onLearnMore: () {
                  onLearnMore?.call(AccountRole.voter);
                },
              );
            },
          )
          .separatedBy(const SizedBox(height: 12))
          .toList(),
    );
```

Reply:
this probably would require adding more methods to  `AccountRoleExt`. See https://github.com/input-output-hk/catalyst-voices/blob/main/catalyst_voices/lib/common/ext/account_role_ext.dart

Reply:
But it also needs to link with its localization and iconography though. I'll create a struct to map the enum with its corresponding localization and iconography then.

Reply:
Its okey, this extension lives in `voices` package and you can access it from there. its common practice. For examples `SpaceExt` has `icon` getter ([link](https://github.com/input-output-hk/catalyst-voices/blob/main/catalyst_voices/lib/common/ext/space_ext.dart))